open-company/open-company-api
# First major API revision

https://trello.com/c/emR4xKsO

Dropped the "reports" concept.

Added the "sections" concept and revision history per section and for the company.

To test:

1. Remove your dev database.
2. Create the database with (db/init) or by `lein start`.
3. Follow the steps in the README to create some companies and exercise them from the Clojure REPL
4. GET those companies with cURL
5. Use the links in the cURL response to GET sections, and older sections (as-of links from revisions)
6. Use cURL to PUT a section update
7. Verify with cURL that updates are in place and new revision has been made
8. Verify with REPL that updates are in place and new revision has been made
9. Use cURL to PATCH a section update
10. Verify with cURL that updates are in place and new revision has been made
11. Verify with REPL that updates are in place and new revision has been made
12. Wipe out your QA database
13. Notice that :commentary is now :notes
13. Run (very limited) tests with `lein test!' and verify they pass.
14. Review this overly large set of code changes (sorry!)

Merge, deploy, eat a twinkie!
